laserchicken/feature_extractor/test_density_extractor.py

```diff
@@ -0,0 +1,188 @@
+import os
+import unittest
+
+import random
+import numpy as np
+
+from laserchicken import keys, read_las, utils
+from laserchicken.compute_neighbors import compute_neighborhoods
+from laserchicken.volume_specification import Sphere, InfiniteCylinder
+from laserchicken.feature_extractor.density_feature_extractor import PointDensityFeatureExtractor
+
+
+class TestDensityFeatureExtractorSphere(unittest.TestCase):
+    """Test density extractor on artificial spherical data."""
+
+    point_cloud = None
+
+    def test_sphere_index(self):
+        """Compute the density for a sphere given as index of the source pc."""
+        neighbors_index = compute_neighborhoods(self.point_cloud,
+                                                self.targetpc,
+                                                self.sphere)
+
+        extractor = PointDensityFeatureExtractor()
+        for index in neighbors_index:
+            d = extractor.extract(self.point_cloud, index, None, None, self.sphere)
+            self.assertEqual(d, self.voldens)
+
+    def _get_central_point(self):
+        """Get the central point."""
+        return utils.copy_pointcloud(self.point_cloud, [0])
+
+    def _get_points_on_sphere(self):
```

Comment:
function name should be different.
get_... suggests that it would return something and doesn't have side effects. It is a bit confusing now when you read the setUp, it comes as a surprise that self.radius is defined, because it was a side effect of this get_... function.
I'd suggest something like:
setup_sphere_data
This suggests that it's not just creating points but might do a bit more, like setting phi and theta etc.